rmc/utils/missense_badness.py

```diff
@@ -230,86 +297,160 @@ def calculate_misbad(
         If False, will read and modify existing output data by adding or modifying columns rather than overwriting entirely.
         If True, will clear existing output data and write new output data.
         The output Table is the missense badness score Table.
+    :param bool use_test_transcripts: Whether to use test transcripts in calculations.
+        If False, will use training transcripts only.
+        If True, will use test transcripts only.
+        Default is False.
+    :param bool do_k_fold_training: Whether to generate k-fold models with the training transcripts.
+        If False, will use all training transcripts in calculation of a single model.
+        If True, will calculate k models corresponding to the k-folds of the training transcripts.
+        Default is False.
+        NOTE that `use_test_transcripts` must be False if `do_k_fold_training` is True.
     :param float oe_threshold: OE Threshold used to split Table.
         Rows with OE less or equal to this threshold will be considered "low" OE, and
         rows with OE greater than this threshold will considered "high" OE.
         Default is 0.6.
-    :param freeze: RMC freeze number. Default is CURRENT_FREEZE.
-    :return: None; writes Table with missense badness score to resource path.
+    :param int freeze: RMC freeze number. Default is CURRENT_FREEZE.
+    :return: None; writes Table(s) with missense badness scores to resource path(s).
     """
-    if not file_exists(amino_acids_oe.versions[freeze].path):
-        raise DataException(
-            "Table with all amino acid substitutions and missense OE doesn't exist!"
+    if use_test_transcripts and do_k_fold_training:
+        raise DataException("Cannot generate k-fold models with test transcripts!")
+
+    if use_test_transcripts or not do_k_fold_training:
+        if not file_exists(
+            amino_acids_oe_path(is_test=use_test_transcripts, freeze=freeze)
+        ):
+            transcript_type = "test" if use_test_transcripts else "training"
+            raise DataException(
+                "Table with all amino acid substitutions and missense OE "
+                f"in {transcript_type} transcripts doesn't exist!"
+            )
+    else:
+        if not all(
+            [
+                file_exists(
+                    amino_acids_oe_path(
+                        is_test=use_test_transcripts,
+                        fold=i,
+                        is_val=is_val,
+                        freeze=freeze,
+                    )
+                )
+                for i in range(1, fold_k + 1)
+                for is_val in [True, False]
+            ]
+        ):
+            raise DataException(
+                "Not all k-fold training and validation tables with all amino acid substitutions "
+                "and missense OE exist!"
```

Comment:
Curious why this is in a separate line

Reply:
Oh just to keep the line under a certain length